File: core/SearchInRotatedSortedArray/SearchInRotatedSortedArray.java
package com.dbc;

public class SearchInRotatedSortedArray {
    public int bSearch(int[] nums, int left, int right, int target) {
        if (left > right) return -1;
        int mid = (left + right) / 2;
        if (nums[mid] == target) return mid;
        int leftNum = bSearch(nums, left, mid - 1, target);
        int rightNum = bSearch(nums, mid + 1, right, target);
        if (leftNum != -1) return leftNum;
        else if (rightNum != -1) return rightNum;
        else return -1;
    }

    public int search(int[] nums, int target) {
        return bSearch(nums, 0, nums.length - 1, target);
    }
}

File: core/SearchInRotatedSortedArray/SearchInRotatedSortedArray.py
from typing import List

def search(nums: List[int], target: int) -> int:
    def b_search(left, right):
        if left > right:
            return -1
        mid = (left + right) // 2
        if nums[mid] == target:
            return mid
        left_num = b_search(left, mid - 1)
        right_num = b_search(mid + 1, right)
        if left_num != -1:
            return left_num
        if right_num != -1:
            return right_num
        return -1

    return b_search(0, len(nums) - 1)


